content/en/BestPractices/SSTI-prevention.md

```diff
@@ -0,0 +1,179 @@
+---
+title: Protect against Server-Side Template Injection
+linkTitle: Protect against SSTI
+weight: 620
+description: >
+  Keep your app safe from SSTI attacks.
+---
+
+{{% pageinfo %}}
+Learn about Server-Side Template Injection (SSTI) exploitation and discover how to prevent them.
+{{% /pageinfo %}}
+
+
+
+SSTI is categorised under injection in the [OWASP Top 10 Web Application Security Risks (2021)](https://owasp.org/www-project-top-ten/) and is ranked 3rd according to its impact and likelihood.
+
+
+## Description
+
+The server-side template injection vulnerability allows an attacker to execute server-side commands by injecting malicious data into a template. A vulnerability occurs when invalid user input is embedded in a template engine. This can result in [remote code execution (RCE)](https://www.imperva.com/learn/application-security/remote-code-execution/).
+
+For example, depending on the IP that accesses a site, the site may look like this:
+
+```ts
+<h1>Welcome to the cobalt!</h1>
+<u>This page is being accessed from the remote address: {{ip}}</u>
+```
+
+It will render the remote address into the {{ip}} variable while reusing the rest of the HTML for every request it receives.
+
+Typically, server-side template injection has a devastating effect. It generally results in remote code execution by completely controlling the back-end server when a server-side template injection vulnerability is exploited. Despite not executing any code, the attacker may still be able to access sensitive server data.
```

Comment:
```suggestion
```html
<h1>Welcome to Example Project!</h1>
<u>You're accessing this page from a remote address: {{ip}}</u>
```

The injection passes a remote address into the `{{ip}}` variable while reusing the rest of the HTML code for every request it receives.

SSTI may have a devastating effect. It generally results in remote code execution by controlling the backend server. Even if the attacker doesn't execute any code, they may still access sensitive server data.
```